src/pybind/mgr/prometheus/module.py

```diff
@@ -889,15 +882,20 @@ def collect(self):
 
         self.get_rbd_stats()
 
-        _end_time = time.time()
-        self.metrics['scrape_duration_seconds'].set(_end_time - _start_time)
-
         # Return formatted metrics and clear no longer used data
         _metrics = [m.str_expfmt() for m in self.metrics.values()]
         for k in self.metrics.keys():
             self.metrics[k].clear()
 
-        return ''.join(_metrics) + '\n'
+        _end_time = time.time()
+        scrape_duration = Metric(
```

Comment:
Why this change? We will always be pushing the `scrape_duration_sec(ond)s` metric so why do we need to create it every time we collect the data?

Reply:
Well we want to set the `scrape_duration` metric after all other metrics have been formatted, so it has to be separate from `self.metrics`

We could also keep this one metric as a member variable of the module. Is there an advantage to that?

Reply:
Well, yes, there is. You don't need to create and destroy the object every time you call collect so there is a (tiny) performance improvement.

Currently, I am looking at `scrape_duration_seconds` provided directly by the prometheus server and I am starting to think that we don't need this metric at all:

https://prometheus.io/docs/concepts/jobs_instances/#automatically-generated-labels-and-time-series

I have hit some weird issues where it said the scraping duration was either 5 or 10 seconds, I am not sure why it did that (I'm still looking into that). It fixed itself after a couple of minutes though.

Reply:
Oh of course...you're right @b-ranto. I'll push a PR to remove it.